### DISCUSSION TITLE
Jhipster sample using DTOs

### DISCUSSION THREAD

**Discussion by istibekesi** [jhipster/generator-jhipster#27928] [Q&A]
I've checked the samlpe projects within this repo, and I have a few questions about the one that says it uses DTOs:
[BankAccountResource.java](https://github.com/jhipster/jhipster-sample-app-dto/blob/bf881889d3c231bd38009e4c6ff57dc44d6ef1dd/src/main/java/io/github/jhipster/sample/web/rest/BankAccountResource.java#L159)

```
    @GetMapping("")
    public List<BankAccountDTO> getAllBankAccounts(
        @RequestParam(name = "eagerload", required = false, defaultValue = "true") boolean eagerload
    ) {
        LOG.debug("REST request to get all BankAccounts");
        List<BankAccount> bankAccounts = bankAccountRepository.findAll();
        return bankAccountMapper.toDto(bankAccounts);
    }
```

**How is this generated, what is the JDL being used? 
It has DTOs, but why it does not have a "BankAccountService" implementation?** 

---
_UPDATE_
I've generated a brand new project with this super simple JDL:
```
entity BankAccount {
	bankAccount String
}

// Use Data Transfer Objects (DTO)
dto * with mapstruct

// Set service options to all except few
// service all with serviceImpl except Employee, Job
```
It generates a BankAccountService and does the mapping in the service layer, not in the controller.
Why the sample project above is different?

**Comment by TatyOko28** [comment]
This is normal in JHipster.

Having DTOs does not automatically mean that JHipster will generate a service class.

In JHipster, DTO generation and service generation are controlled by different JDL options.

For example, this JDL:

```jdl
entity BankAccount {
  name String required
  balance BigDecimal required
}

dto BankAccount with mapstruct
